Sources/performance-worker/ClientStats.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Dispatch
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Musl)
+import Musl
+#elseif canImport(Glibc)
+import Glibc
+#else
+let badOS = { fatalError("unsupported OS") }()
+#endif
+
+#if canImport(Darwin)
+private let OUR_RUSAGE_SELF: Int32 = RUSAGE_SELF
+#elseif canImport(Musl) || canImport(Glibc)
+private let OUR_RUSAGE_SELF: Int32 = RUSAGE_SELF.rawValue
+#endif
```

Comment:
Given we do the same for the server, can we deduplicate all of this? 

We can move the actual `getrusage` call to a static function somewhere (maybe an extension on `System` from `NIOCore`). While we're at it we can make the function a bit easier to use to so that callers don't have to pass in `rusage` and `OUR_RUSAGE_SELF`.

Reply:
Want to make sure I understood right, do you mean the function to be like this?
`extension System {
  static func getUserAndSystemTime() -> (userTime: Double, systemTime: Double) {
    let userTime, systemTime: Double
  
    #if canImport(Darwin)
    let OUR_RUSAGE_SELF: Int32 = RUSAGE_SELF
    #elseif canImport(Musl) || canImport(Glibc)
    let OUR_RUSAGE_SELF: Int32 = RUSAGE_SELF.rawValue
    #endif
    
    var usage = rusage()
    if getrusage(OUR_RUSAGE_SELF, &usage) == 0 {
      // Adding the seconds with the microseconds transformed into seconds to get the
      // real number of seconds as a `Double`.
      userTime = Double(usage.ru_utime.tv_sec) + Double(usage.ru_utime.tv_usec) * 1e-6
      systemTime = Double(usage.ru_stime.tv_sec) + Double(usage.ru_stime.tv_usec) * 1e-6
    } else {
      userTime = 0
      systemTime = 0
    }
    
    return (userTime, systemTime)
  }
}`

Reply:
I was actually thinking we can just return `rusage`:

```
extension System {
  static func resourceUsage() -> rusage {
    // ...
  }
}
```